src/identity-providers.js

```diff
@@ -0,0 +1,95 @@
+'use strict'
+const Identity = require('./identity')
+const OrbitDBIdentityProvider = require('./orbit-db-identity-provider')
+const Keystore = require('orbit-db-keystore')
+const type = 'orbitdb'
+let supportedTypes = {
+  'orbitdb': OrbitDBIdentityProvider,
+}
+
+const getHandlerFor = (type) => {
+  if (!IdentityProviders.isSupported(type)) {
+    throw new Error(`IdentityProvider type '${type}' is not supported`)
+  }
+  return supportedTypes[type]
+}
+
+class IdentityProviders {
+  constructor(options = {}) {
+    this._keystore = options.keystore || Keystore.create(options.keypath || './orbitdb/ipkeys')
+  }
+
+  async sign (identity, data) {
+    const signingKey = await this._keystore.getKey(identity.id)
+    if (!signingKey)
+      throw new Error(`Private signing key not found from Keystore`)
+
+    const signature = await this._keystore.sign(signingKey, data)
+    return signature
+  }
+
+  async verify (signature, publicKey, data) {
+    return this._keystore.verify(signature, publicKey, data)
+  }
+
+  async create(options = {}) {
+    options = Object.assign({}, { type }, options )
+    const IdentityProvider = getHandlerFor(options.type)
+    const identityProvider = new IdentityProvider(options)
+    const id = await identityProvider.createId(options)
+    const { publicKey, idSignature } = await this.signIdentity(id, options)
+    const pubKeyIdSignature = await identityProvider.signIdentity(publicKey + idSignature, options)
+    return new Identity(id, publicKey, idSignature, pubKeyIdSignature, IdentityProvider.type, this)
+  }
+
+  async signIdentity(id, options = {}) {
+    const keystore = options.keystore || this._keystore
```

Comment:
Should we allow the user to pass a keystore here? Is it needed? I feel allowing that allows the user to get confused and pass a _different_ keystore here than in the constructor (and we don't allow passing the keystore in verifyIdentity)